### DISCUSSION TITLE
Gradle: Support external file in "apply from" statement

### DISCUSSION THREAD

**Discussion by MalachowskiTomasz** [renovatebot/renovate#22088] [Suggest an Idea]
### Type of discussion.

I'm proposing an idea

### Tell us more.

Currently, renovate bot works only with gradle files that are stored in a project. However, in our projects we rely on shared configuration uploaded to publicly available website containing info about gradle registries. Renovate is not able to pick up that configuration thus it was unusable in our use-case...

There was [a PR](https://github.com/renovatebot/renovate/pull/22059) opened but based on discussion we've decided to talk about it first here. Feel free to add your comments/suggestions

**Comment by rarkins** [comment]
We prefer to avoid doing "lookups" during the scan/extract phase. We are not needing to do that for Gradle or any other manager today.

I'll leave @Churro to answer next whether this is "best practice" we need to support regardless, or whether we recommend any different approach.

**Comment by Churro** [comment]
Besides the fact that synchronously sideloading and parsing untrusted code from the Internet renders renovate execution indeterministic and security-wise concerning, I'm not convinced if it would actually make sense for gradle manager: 

A feature as suggested in PR #22059 would make renovate lookup updates for dependencies declared in remote files too. As those files aren't available locally, renovate couldn't open PRs for them.

In common use cases, package registries remain rather stable. If not defined in the local repository, you could still make renovate use them via package rules (example [here](https://docs.renovatebot.com/configuration-options/#registryurls)). In other words, there is a workaround for your use-case that is safe from potential side effects.
